### PR TITLE
Fix small bugs in Android OS Plugin

### PR DESCRIPTION
**Pull request by MaxGroot**
* Fix incorrect accessing of path on the target's filesystem
* Fix exceptions when build property doesn't contain a security patch
* Add small test

### CODE REVIEW

File: dissect/target/plugins/os/unix/linux/android/_os.py [hunk 1]
```diff
@@ -50,9 +50,17 @@ def ips(self) -> list[str]:
...
     @export(property=True)
     def version(self) -> str:
-        release_version = self.props.props["ro.build.version.release"]
-        security_patch_version = self.props.props["ro.build.version.security_patch"]
-        return f"Android {release_version} ({security_patch_version})"
+        full_version = "Android"
+
+        release_version = self.props.props.get("ro.build.version.release")
+        if release_version:
+            full_version += f" {release_version}"
```
Comment (Schamper) [inline dissect/target/plugins/os/unix/linux/android/_os.py:57]:
```suggestion
        if release_version := self.props.props.get("ro.build.version.release"):
            full_version += f" {release_version}"
```

File: dissect/target/plugins/os/unix/linux/android/_os.py [hunk 2]
```diff
@@ -50,9 +50,17 @@ def ips(self) -> list[str]:
...
+
+        security_patch_version = self.props.props.get("ro.build.version.security_patch")
+        if security_patch_version:
+            full_version += f" ({security_patch_version})"
```
Comment (Schamper) [inline dissect/target/plugins/os/unix/linux/android/_os.py:61]:
```suggestion
        if security_patch_version := self.props.props.get("ro.build.version.security_patch"):
            full_version += f" ({security_patch_version})"
```

File: tests/conftest.py [hunk 3]
```diff
@@ -258,6 +258,19 @@ def target_citrix(fs_bsd):
     yield mock_target
 
 
+@pytest.fixture
+def target_android() -> Target:
+    mock_target = next(make_mock_target())
```
Comment (Schamper) [inline tests/conftest.py:263]:
```suggestion
    mock_target = next(make_mock_target())
    mock_target._os_plugin = AndroidPlugin
```

File: tests/conftest.py [hunk 4]
```diff
@@ -258,6 +258,19 @@ def target_citrix(fs_bsd):
...
+
+    mock_filesystem = VirtualFilesystem()
+    mock_filesystem.map_file("/build.prop", absolute_path("data/plugins/os/unix/linux/android/build.prop"))
```
Comment (Schamper) [inline tests/conftest.py:266]: Could you split this into a separate fixture called `fs_android`?

File: tests/test_plugins_os_unix_linux_android.py [hunk 5]
```diff
@@ -0,0 +1,9 @@
+from dissect.target.plugins.os.unix.linux.android._os import AndroidPlugin
+
+
+def test_unix_linux_android_os(target_android):
```
Comment (Schamper) [inline tests/test_plugins_os_unix_linux_android.py:4]:
```suggestion
def test_android_os(target_android: Target) -> None:
```

**Review by Schamper — CHANGES_REQUESTED** [review]

**Comment by MaxGroot** [comment]
While testing on a different Android system I found some other inconsistencies, I'll make sure to add fixes for those in this PR as well. In the meantime I'm converting it back to draft.

File: tests/conftest.py [hunk 6]
```diff
@@ -259,14 +267,10 @@ def target_citrix(fs_bsd):
 
 
 @pytest.fixture
-def target_android() -> Target:
+def target_android(fs_android) -> Target:
```
Comment (Schamper) [inline tests/conftest.py:270]:
```suggestion
def target_android(fs_android: VirtualFilesystem) -> Target:
```

File: tests/test_plugins_os_unix_linux_android.py [hunk 7]
```diff
@@ -1,7 +1,7 @@
 from dissect.target.plugins.os.unix.linux.android._os import AndroidPlugin
 
 
-def test_unix_linux_android_os(target_android):
+def test_android_os(target_android) -> None:
```
Comment (Schamper) [inline tests/test_plugins_os_unix_linux_android.py:4]:
```suggestion
def test_android_os(target_android: Target) -> None:
```

**Review by Schamper — CHANGES_REQUESTED** [review]

**Review by Schamper — APPROVED** [review]

**Pull request merged**